apps/andi/test/unit/andi_web/live/edit_live_view_test.exs

```diff
@@ -222,86 +249,61 @@ defmodule AndiWeb.EditLiveViewTest do
       assert {:ok, view, html} = live(conn, @url_path <> dataset.id)
       assert get_select(html, "#metadata_private") == {"true", "Private"}
 
-      dataset_map = dataset_to_map(dataset) |> Map.put(:private, false)
+      dataset_map = dataset_to_form_data(dataset) |> Map.put(:private, false)
 
       html = render_change(view, :validate, %{"metadata" => dataset_map})
       assert get_select(html, "#metadata_private") == {"false", "Public"}
     end
 
-    test "All required fields display proper error message", %{conn: conn} do
-      assert_error_message(
-        conn,
-        TDG.create_dataset(%{business: %{dataTitle: ""}}),
-        :dataTitle,
-        "Please enter a valid dataset title."
-      )
-
-      assert_error_message(
-        conn,
-        TDG.create_dataset(%{business: %{description: ""}}),
-        :description,
-        "Please enter a valid description."
-      )
-
-      assert_error_message(
-        conn,
-        TDG.create_dataset(%{business: %{contactName: ""}}),
-        :contactName,
-        "Please enter a valid maintainer name."
-      )
-
-      assert_error_message(
-        conn,
-        TDG.create_dataset(%{business: %{contactEmail: ""}}),
-        :contactEmail,
-        "Please enter a valid maintainer email."
-      )
-
-      assert_error_message(
-        conn,
-        TDG.create_dataset(%{business: %{issuedDate: ""}}),
-        :issuedDate,
-        "Please enter a valid release date."
-      )
-
-      assert_error_message(
-        conn,
-        TDG.create_dataset(%{business: %{license: ""}}),
-        :license,
-        "Please enter a valid license."
-      )
+    data_test "required #{field} field displays proper error message", %{conn: conn} do
+      html = save_form_for_dataset(conn, TDG.create_dataset(dataset_override))
 
+      assert get_text(html, "##{field}-error-msg") == expected_error_message
+
+      where([
+        [:field, :dataset_override, :expected_error_message],
```

Comment:
should some of these be using the display names from within apps/andi/lib/andi/input_schemas/display_names.ex?

Reply:
They are using those names.  We just don't want to reference that module in the test so changes to these messages still has to be intentional.